scripts/files/fs_s3.py

```diff
@@ -0,0 +1,100 @@
+from scripts.logging.time_helper import time_in_ms
+
+if TYPE_CHECKING:
+    from mypy_boto3_s3.service_resource import S3ServiceResource
+else:
+    S3ServiceResource = object
+
+# s3_session = [{"bucket", boto3.Session}]
+s3_sessions: Dict[str, S3ServiceResource] = {}
+
+
+def _get_s3_resource(bucket_name: str) -> S3ServiceResource:
+    """Return a boto3 S3 Resource with the AWS credentials for a bucket.
+
+    Args:
+        bucket_name (str): The name of the bucket.
+
+    Returns:
+        S3ServiceResource: The boto3 S3 Resource.
+    """
+    session = s3_sessions.get(bucket_name, None)
+    if session is None:
+        # TODO implement refreshable session TDE-235
+        credentials = get_credentials(bucket_name)
+        session = boto3.Session(
+            aws_access_key_id=credentials.access_key,
+            aws_secret_access_key=credentials.secret_key,
+            aws_session_token=credentials.token,
+        )
+    s3_sessions[bucket_name] = session
+
+    s3: S3ServiceResource = session.resource("s3")
+    return s3
+
+
+def write(destination: str, source: bytes) -> None:
+    """Write a source (bytes) in a AWS s3 destination (path in a bucket).
+
+    Args:
+        destination (str): The AWS S3 path to the file to write.
+        source (bytes): The source file in bytes.
+    """
+    start_time = time_in_ms()
+    if source is None:
+        get_log().error("write_s3_source_none", path=destination, error="The 'source' is None.")
+        raise Exception("The 'source' is None.")
+    s3_path = parse_path(destination)
+    key = s3_path.key
+    s3 = boto3.resource("s3")
+
+    try:
+        s3_object = s3.Object(s3_path.bucket, key)
+        s3_object.put(Body=source)
+        get_log().debug("write_s3_success", path=destination, duration=time_in_ms() - start_time)
+    except botocore.exceptions.ClientError as ce:
+        get_log().error("write_s3_error", path=destination, error=f"Unable to write the file: {ce}")
+        raise ce
+
+
+def read(path: str, needs_credentials: bool = False) -> bytes:
+    """Read a file on a AWS S3 bucket.
+
+    Args:
+        path (str): The AWS S3 path to the file to read.
+        needs_credentials (bool, optional):  Tells if credentials are needed. Defaults to False.
+
+    Returns:
+        bytes: The file in bytes.
+    """
+    start_time = time_in_ms()
+    s3_path = parse_path(path)
+    key = s3_path.key
+    s3 = boto3.resource("s3")
+
+    try:
+        if needs_credentials:
+            s3 = _get_s3_resource(s3_path.bucket)
+
+        s3_object = s3.Object(s3_path.bucket, key)
+        file: bytes = s3_object.get()["Body"].read()
+    except botocore.exceptions.ClientError as ce:
+        if not needs_credentials and ce.response["Error"]["Code"] == "AccessDenied":
+            get_log().debug("read_s3_needs_credentials", path=path)
+            return read(path, True)
+        if ce.response["Error"]["Code"] == "NoSuchBucket":
```

Comment:
you should use the `except boto...exceptions....NoSuchBucket` style for this

Reply:
Ha, I removed my comment somehow. When using boto3 `Resources`, exceptions are a bit different than for `Client`. 
https://boto3.amazonaws.com/v1/documentation/api/latest/guide/error-handling.html#catching-exceptions-when-using-a-resource-client
I am not able to find the `AccessDenied` exception. Should I keep the existing behavior for `AccessDenied` and catch the exception using `except` on each specific exceptions for the other errors?